### DISCUSSION TITLE
How does firebase.json config work in ios

### DISCUSSION THREAD

**Discussion by aliraza-noon** [invertase/react-native-firebase#5524] [General]
hello there, 

I have been trying to test my ios crashlytics  integration for but unfortunately it did not really work, I could not get value of  `crashlytics_debug_enabled` to set to true I was always getting false for final value.

I started to dig into code to understand it better I am no objective-c expert but `RNFBJSON` implementation seems wrong all the methods on singleton reference `_firebaseJson` property, in my local `node_modules/@react-native-firebase/app/ios/RNFBApp/RNFBJSON.m` I changed for instance `contains` method 

from
```
- (BOOL)contains:(NSString *)key {
  return [_firebaseJson valueForKey:key] != nil;
}
```
to
```
- (BOOL)contains:(NSString *)key {
  RNFBJSON *instance = [RNFBJSON shared];
  return [instance.firebaseJson valueForKey:key] != nil;
}
```

it started working I am wondering if this should be changed in the project as well ?

**Comment by mikehardy** [comment]
Have you consulted this? https://invertase.io/blog/react-native-firebase-crashlytics-configuration

Here's a deep link to the last time I was in the area (doing the work that resulted in that blog post) that goes to the heart of the iOS side of the config from firebase.json, it should be a good handle to explore from https://github.com/invertase/react-native-firebase/pull/5147/files#diff-d164085f74fc0d86d9a9a1ed45b2f8e12335939192741e4ffe24c8d538d5b7b9

**Reply by aliraza-noon** [reply]
@mikehardy I did follow your article exactly, but I was not able to test ios integration because I was always getting `crashlytics_debug_enabled` as `false` maybe I am missing something here as result crashes were not being reported in console

**Reply by mikehardy** [reply]
There must be something unexpected in your project - you *were* able to test it, you just got an unexpected result yes? Scientific process is frustrating like that sometimes. All I can say is that it definitely works in iOS in debug or at least it did in all the testing I performed while implementing the last batch of features there and documenting it all for that article. Keep digging and hopefully you find it
